### PR TITLE
Diffs in V 0.7.0 release candidate with respect to develop

### PR DESCRIPTION
**Pull request by bmhan12**
This PR:
- A helper PR to #916 to make it easier to see changes with respect to develop.
- Add comments and review to #916, use this PR for viewing.

### CODE REVIEW

File: RELEASE [hunk 1]
```diff
@@ -20,15 +20,16 @@ The Axom development team is comprised of:
...
-Esteban Pauli (pauli2@llnl.gov)
 Lee Taylor (taylor16@llnl.gov)
 Kenny Weiss (kweiss@llnl.gov)
 Chris White (white238@llnl.gov)
 Max Yang (wang39@llnl.gov)
 
 emeritus
 George Zagaris
+Josh Essman
+Pauli Esteban
```
Comment (kennyweiss) [inline RELEASE:35]:
```suggestion
Esteban Pauli
```

File: RELEASE [hunk 2]
```diff
@@ -20,15 +20,16 @@ The Axom development team is comprised of:
...
-Esteban Pauli (pauli2@llnl.gov)
 Lee Taylor (taylor16@llnl.gov)
 Kenny Weiss (kweiss@llnl.gov)
 Chris White (white238@llnl.gov)
 Max Yang (wang39@llnl.gov)
 
 emeritus
 George Zagaris
+Josh Essman
+Esteban Pauli
```
Comment (rhornung67) [inline RELEASE:35]: Since we're maintaining these lists of names, we should also include Randy Settgast, Aaron Black, and Ben Corbett as emeriti.

Comment (rhornung67) [reply RELEASE:35]: Also, should we add a list of summer students who have contributed over the years, after all the code they developed eventually made it in? @kennyweiss at al. thoughts?

File: RELEASE [hunk 3]
```diff
@@ -16,19 +16,4 @@ Unlimited Open Source - BSD Distribution
...
-Rich Hornung (hornung1@llnl.gov)
-Esteban Pauli (pauli2@llnl.gov)
-Lee Taylor (taylor16@llnl.gov)
-Kenny Weiss (kweiss@llnl.gov)
-Chris White (white238@llnl.gov)
-Max Yang (wang39@llnl.gov)
-
-emeritus
-George Zagaris
+See https://github.com/LLNL/axom/contributors for the list of contributors.
```
Comment (kennyweiss) [inline RELEASE:19]: 👍

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -19,6 +19,8 @@ The Axom project release numbers follow [Semantic Versioning](http://semver.org/
 
 ## [Unreleased] - Release date yyyy-mm-dd
 
+## [Version 0.7.0] - Release date 2022-8-30
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:22]: Minor: Should this be `08` ?

```suggestion
## [Version 0.7.0] - Release date 2022-08-30
```

**Pull request closed without merging**